packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -409,12 +407,14 @@ class RenderParagraph extends RenderBox
     while (child != null) {
       final TextParentData textParentData = child.parentData;
       final double scale = textParentData.scale;
-      final bool isHit = result.addWithPaintOffset(
-        offset: textParentData.offset,
+      Matrix4 localPosTrans = Matrix4.translation(Vector3(textParentData.offset.dx, textParentData.offset.dy, 0));
+      Matrix4 scaleTrans = Matrix4.diagonal3Values(scale, scale, scale);
+      Matrix4 transform = localPosTrans * scaleTrans;
+      final bool isHit = result.addWithPaintTransform(
+        transform: transform,
         position: position,
         hitTest: (BoxHitTestResult result, Offset transformed) {
-          assert(transformed == position - textParentData.offset);
-          return child.hitTest(result, position: transformed / scale);
+          return child.hitTest(result, position: transformed);
```

Comment:
It's probably a good idea to leave the assert in here to ensure that the matrix actually does what we expect.